src/main/java/com/hillel/items_exchange/exceptionshandling/GlobalControllerExceptionHandler.java

```diff
@@ -0,0 +1,137 @@
+
+        log.warn(errorMessage.errorLog());
+
+        return new ResponseEntity<>(errorMessage, HttpStatus.NOT_FOUND);
+    }
+
+    @ExceptionHandler(IllegalIdentifierException.class)
+    public ResponseEntity<Object> handleIllegalIdentifierException(IllegalIdentifierException e,
+                                                                   ServletWebRequest request) {
+
+        ErrorMessage errorMessage = new ErrorMessage(new Date(), HttpStatus.BAD_REQUEST.value(),
+                messageSourceUtil.getExceptionMessageSource("illegal.id.exception"),
+                e.getMessage(), request.getRequest().getRequestURI(), request.getHttpMethod());
+
+        log.warn(errorMessage.errorLog());
+
+        return new ResponseEntity<>(errorMessage, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(InvalidDtoException.class)
+    public ResponseEntity<Object> handleInvalidDtoException(InvalidDtoException e,
+                                                            ServletWebRequest request) {
+
+        ErrorMessage errorMessage = new ErrorMessage(new Date(), HttpStatus.BAD_REQUEST.value(),
+                messageSourceUtil.getExceptionMessageSource("invalid.dto.exception"),
+                e.getMessage(), request.getRequest().getRequestURI(), request.getHttpMethod());
+
+        log.warn(errorMessage.errorLog());
+
+        return new ResponseEntity<>(errorMessage, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(DataIntegrityViolationException.class)
+    public ResponseEntity<Object> handleSqlException(DataIntegrityViolationException e,
+                                                     ServletWebRequest request) {
+
+        ErrorMessage errorMessage = new ErrorMessage(new Date(), HttpStatus.BAD_REQUEST.value(),
+                messageSourceUtil.getExceptionMessageSource("sql.exception"),
+                e.getLocalizedMessage(), request.getRequest().getRequestURI(), request.getHttpMethod());
+
+        log.error(errorMessage.errorLog());
+
+        return new ResponseEntity<>(errorMessage, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<Object> handleIllegalArgumentException(IllegalArgumentException e,
+                                                                 ServletWebRequest request) {
+
+        ErrorMessage errorMessage = new ErrorMessage(new Date(), HttpStatus.BAD_REQUEST.value(),
+                messageSourceUtil.getExceptionMessageSource("illegal.argument.exception"),
+                e.getLocalizedMessage(), request.getRequest().getRequestURI(), request.getHttpMethod());
+
+        log.warn(errorMessage.errorLog());
+
+        return new ResponseEntity<>(errorMessage, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(ConstraintViolationException.class)
+    public ResponseEntity<Object> handleConstraintViolationException(ConstraintViolationException e,
```

Comment:
Could you remind me when we get this one exception?

Reply:
When JSR-303 validation failed